adminSiteServer/apiRouter.ts

```diff
@@ -2271,24 +2271,40 @@ apiRouter.get("/posts/:postId.json", async (req: Request, res: Response) => {
 
 apiRouter.post("/posts/:postId/createGdoc", async (req: Request) => {
     const postId = expectInt(req.params.postId)
+    const allowRecreate = !!req.body.allowRecreate
     const post = (await db
         .knexTable(postsTable)
         .where({ id: postId })
         .select("*")
         .first()) as PostRow | undefined
+
     if (!post) throw new JsonError(`No post found for id ${postId}`, 404)
-    if (post.gdocSuccessorId)
+    const existingGdocId = post.gdocSuccessorId
+    if (!allowRecreate && existingGdocId)
         throw new JsonError("A gdoc already exists for this post", 400)
     const archieMl = JSON.parse(post.archieml) as OwidArticleType
-    const gdocId = await createGdocAndInsertOwidArticleContent(archieMl.content)
-    post.gdocSuccessorId = gdocId
-    await db
-        .knexTable(postsTable)
-        .where({ id: postId })
-        .update("gdocSuccessorId", gdocId)
-    // TODO: fill the rest of content required for an entry in the gdoc table,
-    // mark the post in the posts table as no-longer editable/publish-able,
-    // publish through gdocs?
+    const gdocId = await createGdocAndInsertOwidArticleContent(
+        archieMl.content,
+        post.gdocSuccessorId
+    )
+    if (!existingGdocId) {
```

Comment:
What happens if there _is_ an `existingGdocId` ? Shouldn't we still be updating the DB?

Reply:
In this case the posts table already contains a link to this gdoc document and we only needed to regenerate the gdoc content. I've added a comment to make this more clear